### PR TITLE
ng bootstrap removal for consignment feature dialog (CXSPA-1248)

### PR DESCRIPTION
**Pull request by psmul**
Closes: CXSPA-1248

### CODE REVIEW

File: projects/storefrontstyles/scss/cxbase/blocks/modal.scss [hunk 1]
```diff
@@ -37,37 +37,71 @@ $modal-min-width-sm: 100% !important;
   }
 }
 
-.modal-xl,
-.modal-lg {
+// Ng-bootstrap modal Styling
```
Comment (morganm58) [inline projects/storefrontstyles/scss/cxbase/blocks/modal.scss]: You can remove these commented out styles

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html [hunk 2]
```diff
@@ -1,145 +1,159 @@
...
+      <!-- Modal Body -->
+      <!-- shipment header -->
+      <ng-container
+        *ngIf="
+          consignmentTracking?.carrierDetails &&
+            consignmentTracking?.trackingID;
+          else noTracking
+        "
+      >
+        <div class="shipment-heading">
```
Comment (morganm58) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html]: Should have the `cx-` prefix on the classes

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html [hunk 3]
```diff
@@ -1,145 +1,159 @@
...
+          <div class="row">
+            <div class="col-sm-12 col-md-6">
+              <div class="shipment-title">
```
Comment (morganm58) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html]: Same here with the `cx-` prefix on the classes

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html [hunk 4]
```diff
@@ -1,145 +1,159 @@
...
+                {{
+                  'orderDetails.consignmentTracking.dialog.shipped'
+                    | cxTranslate
+                }}
+              </div>
+              <div class="shipment-content">
```
Comment (morganm58) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html]: same here with the `cx` prefix here as well

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html [hunk 5]
```diff
@@ -1,145 +1,159 @@
...
-          <div class="col-sm-12 col-md-6">
-            <div class="shipment-title">
-              {{
-                'orderDetails.consignmentTracking.dialog.carrier' | cxTranslate
-              }}
-            </div>
-            <div class="shipment-content">
-              {{ consignmentTracking?.carrierDetails?.name }}
+      <!-- tracking events -->
+      <div class="events modal-body">
```
Comment (morganm58) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.html]: `cx-` prefix on `events` as well and the other event classes below

File: docs/migration/5_0.md [hunk 6]
```diff
@@ -515,3 +515,28 @@ Due to i18next migration, certain set of keys have been migrated (from `_plural`
...
+- changed detection strategy to `OnPush`
+- Removed `NgbActiveModal` from constructor
+- Added `LaunchDialogService` to constructor
+- `close` is now using `closeDialog` method from `LaunchDialogService` to control dialog dismissing behaviour
+
+#### Template changes 
+- All classes related to component templating have been prefixes with `cx-`
+- Changed `<div class="events modal-body">` to `<div class="cx-tracking-events modal-body">`
+- Changed `<div class="event-body">` to `<div class="cx-tracking-event-body">`
+- Changed `<div class="event-content">` to `<div class="-tracking-event-content">`
```
Comment (pawelfras) [inline docs/migration/5_0.md]: Missing `cx` in class name

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/consignment-tracking.component.ts [hunk 7]
```diff
@@ -45,19 +62,25 @@ export class ConsignmentTrackingComponent implements OnInit, OnDestroy {
...
+
+    const dialog = this.launchDialogService.openDialog(
+      LAUNCH_CALLER.CONSIGNMENT_TRACKING,
+      this.element,
+      this.vcr,
+      modalInstanceData
+    );
+
+    if (dialog) {
+      this.subscription.add(dialog.pipe(take(1)).subscribe());
```
Comment (pawelfras) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/consignment-tracking.component.ts]: `query:` Do we need to use `subscription` in this case? I believe `take(1)` is handling unsubscribe under the hood. Consider to remove `Subscription` and `OnDestroy` interface.

File: feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.ts [hunk 8]
```diff
@@ -1,23 +1,54 @@
...
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
 
   constructor(
-    public activeModal: NgbActiveModal,
-    private orderHistoryFacade: OrderHistoryFacade
+    private orderHistoryFacade: OrderHistoryFacade,
+    protected launchdialogService: LaunchDialogService
```
Comment (pawelfras) [inline feature-libs/order/components/order-details/order-detail-items/consignment-tracking/tracking-events/tracking-events.component.ts]: Missing camelCase

**Review by pawelfras — CHANGES_REQUESTED** [review]

**Review by pawelfras — APPROVED** [review]

**Pull request merged**